### PR TITLE
Feature/author view

### PR DESCRIPTION
**Pull request by pepemaluza**
# Description

Please include a summary of the change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How has this been tested?

Please describe the tests that you ran to verify your changes.

- [ ] Automation testing (Unit or integration tests)
- [x ] Manual testing (Postman)

# Checklist:

- [x ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x ] My pull request is up to date with develop

### CODE REVIEW

File: src/scenes/main/Author/Author.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+                    loading={getAuthorDataStatus === RequestStatus.LOADING}
+                    error={getAuthorDataStatus === RequestStatus.ERROR}
+                />
+            );
+        }
+    }
+
+    renderButtons() {
+        const {isAdmin} = this.state;
+        if (!isAdmin)
```
Comment (pepemaluza) [inline src/scenes/main/Author/Author.tsx:176]: No tiene que estar negado.

File: src/services/AuthorService.ts [hunk 2]
```diff
@@ -24,6 +26,42 @@ const createAuthor = (author: NewAuthor, photo: File): Promise<AxiosResponse<Aut
...
+}
+
+
+const getAuthorData = (authorID: AuthorID): Promise<AxiosResponse<Author>> => {
+    const config: AxiosRequestConfig = {
+        headers: {
+            'Content-Type': undefined,
+        }
+    }
+    return instance.post<Author>(`${baseURL}/authors/${authorID}`, config)
```
Comment (matiasvenditti) [inline src/services/AuthorService.ts:49]: No funciona el GET, esto debería ser: 
return instance.post<Author>(`${baseURL}/authors/${**authorID.id**}`, config)

File: src/services/AuthorService.ts [hunk 3]
```diff
@@ -24,6 +26,42 @@ const createAuthor = (author: NewAuthor, photo: File): Promise<AxiosResponse<Aut
...
+        headers: {
+            'Content-Type': undefined,
+        }
+    }
+    return instance.post<Author>(`${baseURL}/authors/update/${author.id}`, changeAuthorForm, config)
+}
+
+
+const getAuthorData = (authorID: AuthorID): Promise<AxiosResponse<Author>> => {
+    const config: AxiosRequestConfig = {
```
Comment (matiasvenditti) [inline src/services/AuthorService.ts:44]: No hace falta pasarle esta config porque el content type de la request es json (application/json), y es el default para axios, asi que no haría falta.

File: src/services/AuthorService.ts (hunk 2 above)
Comment (matiasvenditti) [inline src/services/AuthorService.ts:49]: Debería ser **instance.get**

File: src/scenes/main/Author/Author.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    /**
+     * Renders AuthorView or AuthorEdit
+     */
+
+    renderAuthor() {
+        const {editAuthorMode, data, books, getAuthorDataStatus} = this.state;
+        if (getAuthorDataStatus === RequestStatus.LOADING) {
+            return (
+                <div>
+                    <Typography align='center'> <Loader /> </Typography>
```
Comment (matiasvenditti) [inline src/scenes/main/Author/Author.tsx:158]: Aca le tuve que poner a Typography `variant='subtitle1'` para sacar un error de la consola

File: src/scenes/main/Author/Author.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+                </div>
+            );
+        }
+        if (editAuthorMode) {
+            return (
+                <AppBar position='static'>
```
Comment (matiasvenditti) [inline src/scenes/main/Author/Author.tsx:164]: AppBar no hace falta ponerlo porque se agrega en Router siempre

File: src/scenes/main/Author/Author.tsx [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+                <ModifyAuthorForm
+                    data={data}
+                    onCancel={this.handleCancel}
+                    onSubmit={this.modifyAuthor}
+                />
+                </AppBar>
```
Comment (matiasvenditti) [inline src/scenes/main/Author/Author.tsx:170]: AppBar no va

File: src/scenes/main/Author/Author.tsx [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+                .catch((error) => {
+                    this.setState({...this.state, updateStatus: RequestStatus.ERROR, error});
+                    this.props.editAuthorCallback(RequestStatus.ERROR);
+                })
+    }
+
+    render() {
+        return (
+            <div className='route-container'>
+                <div className='card-container'>
```
Comment (matiasvenditti) [inline src/scenes/main/Author/Author.tsx:138]: Para que esto te quede mas parecido a lo que es el wireframe yo le cambiaria la clase del `main.css` en la linea 8.

```
.route-container .card-container {
    width: 100%;
    box-sizing: border-box;
    padding: 20px 65px 15px;
    margin-top: 20px;
    /* background-color: #dddddd; */
}
```

**Review by matiasvenditti — CHANGES_REQUESTED** [review]
Dejo dos cambios simples

1. En Author.tsx es un fix de un error en la terminal + sacar el AppBar que no hace falta
2. En main.css un fix de estilos

**Review by matiasvenditti — APPROVED** [review]

**Pull request merged**